hi-ml-azure/testazure/testazure/test_himl.py

```diff
@@ -1755,6 +1765,36 @@ def after_submission(job: Job, ml_client: MLClient) -> None:
     assert after_submission_called, "after_submission callback was not called"
 
 
+def test_submitting_script_with_sdk_v2_passes_display_name(tmp_path: Path) -> None:
+    """
+    Test that submission of a script with SDK v2 passes the display_name parameter to the "command" function
+    that does the actual submission
+    """
+    # Create a minimal script in a temp folder.
+    test_script = tmp_path / "test_script.py"
+    test_script.write_text("print('hello world')")
+    shared_config_json = get_shared_config_json()
+    conda_env_path = create_empty_conda_env(tmp_path)
+    display_name = "my_display_name"
+
+    with check_config_json(tmp_path, shared_config_json=shared_config_json),\
+            change_working_directory(tmp_path), \
+            patch("health_azure.himl.command", side_effect=ValueError) as mock_command:
+        with pytest.raises(ValueError):
+            himl.submit_to_azure_if_needed(
+                aml_workspace=None,
+                entry_script=test_script,
+                conda_environment_file=conda_env_path,
+                snapshot_root_directory=tmp_path,
+                submit_to_azureml=True,
+                strictly_aml_v1=False,
```

Comment:
It would be great to test sdk v1 as well. Mock submit run? and make sure display_name is part of  call_kwargs?

Reply:
V1 is tested too, around line 1064 in `test_himl.py`